lib/enmeshed/connector_api.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Enmeshed
+  class ConnectorApi
+    def initialize(connector_base_url, api_key)
+      @base_url = connector_base_url
+      @api_key = api_key
+      @enmeshed_address = nil
+    end
+
+    def enmeshed_address
+      return @enmeshed_address if @enmeshed_address.present?
+
+      response = Net::HTTP.get(URI("#{@base_url}/api/v2/Account/IdentityInfo"), {'X-API-KEY' => @api_key})
```

Comment:
We haven't extracted the `/api/v2` part here, yet. However, since we introduced Faraday, we construct URLs differently (only passing the path, not the hostname). Thus, extracting is not necessary any longer.